qutip/td_qobj.py

```diff
@@ -0,0 +1,1111 @@
+# This file is part of QuTiP: Quantum Toolbox in Python.
+#
+#    Copyright (c) 2011 and later, Paul D. Nation and Robert J. Johansson.
+#    All rights reserved.
+#
+#    Redistribution and use in source and binary forms, with or without
+#    modification, are permitted provided that the following conditions are
+#    met:
+#
+#    1. Redistributions of source code must retain the above copyright notice,
+#       this list of conditions and the following disclaimer.
+#
+#    2. Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#
+#    3. Neither the name of the QuTiP: Quantum Toolbox in Python nor the names
+#       of its contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+#    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+#    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#    HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+"""Time-dependent Quantum Object (Qobj) class.
+"""
+
+from qutip import Qobj
+from scipy.interpolate import CubicSpline
+from functools import partial
+from types import FunctionType, BuiltinFunctionType
+import numpy as np
+from numbers import Number
+from qutip.superoperator import liouvillian, lindblad_dissipator
+from qutip.td_qobj_codegen import _compile_str_single, make_united_f_ptr, make_united_f_ptr_spline
+from qutip.cy.spmatfuncs import (cy_expect_rho_vec, cy_expect_psi, spmv)
+from qutip.cy.td_qobj_cy import cy_cte_qobj, cy_td_qobj
+
+
+class td_Qobj:
+    """A class for representing time-dependent quantum objects,
+    such as quantum operators and states.
+
+    The td_Qobj class is a representation of time-dependent Qutip quantum
+    objects (Qobj). This class implements math operations :
+        +,- : td_Qobj, Qobj
+        * : Qobj, C-number
+        / : C-number
+    and some common linear operator/state operations. The td_Qobj
+    are constructed from a nested list of Qobj with their time-dependent
+    coefficients.
+
+    *Examples*
+
+        H = [H0, [H1, 'sin(w*t)'], [H2, f1_t], [H3, np.sin(3*w*tlist)]]
```

Comment:
As it stands now, we do not support mixing str and Python func formats.  Of course we could, but then all the solvers would need to add this support.

Reply:
While the different format work together, there are not many advantages to mix then. In some methods, there are optimizations for only func format or only str format.  Should I just remove it from the documentation?